apps/prairielearn/python/prairielearn/internal/question_phases.py

```diff
@@ -143,7 +144,21 @@ def process_element(
             temp_tail = element.tail
             element.tail = None
 
-            element_value = mod[phase](lxml.html.tostring(element), data)
+            args = [lxml.html.tostring(element), data]
+
+            # We need to support legacy element functions, which take three arguments.
+            # The second argument is `element_index`; we'll pass `None`. This is
+            # consistent with the same backwards-compatibility logic in `zygote.py`.
+            arg_names = list(signature(mod[phase]).parameters.keys())
+            if (
```

Comment:
```suggestion
            if arg_names == ["element_html", "element_index", "data"]:
```
Not recommending you do this here for consistency with the zygote, but I think this shortening works.

Reply:
I like that a lot better, I tested it and indeed it works as expected. Changed in cd7f5eb.